src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobTask.cs

```diff
@@ -60,7 +60,7 @@ public class ExportJobTask : IExportJobTask
             IExportDestinationClient exportDestinationClient,
             IResourceDeserializer resourceDeserializer,
             IScoped<IAnonymizerFactory> anonymizerFactory,
-            IMediator mediator,
+            Func<IScoped<IMediator>> mediatorFactory,
```

Comment:
Thought/question: if the parent scope is disposed, would the child scope also be disposed here?

Reply:
I don't know to be honest. :( ChatGPT says:

> No, if the parent IServiceProvider is disposed, the child scope created with CreateScope will not be disposed automatically. You need to dispose of the child scope explicitly.

When prompted further:
> When you create a scope using CreateScope() on an IServiceProvider, the created scope is independent of the parent IServiceProvider. If the parent IServiceProvider is disposed, it does not automatically dispose of the child scopes. Each scope created using CreateScope() needs to be explicitly disposed of to ensure proper cleanup of any IDisposable services that were resolved within that scope.

> This behavior is confirmed in the .NET dependency injection guidelines and various discussions on Stack Overflow​ ([Stack Overflow](https://stackoverflow.com/questions/43244316/iserviceprovider-garbage-collection-disposal))​​ ([MS Learn](https://learn.microsoft.com/en-us/dotnet/core/extensions/dependency-injection-guidelines))​​ ([Stack Overflow](https://stackoverflow.com/questions/70864055/call-iservicescopefactory-createscope-in-a-scoped-service))​. For more information, you can refer to the official Microsoft documentation on dependency injection.

Reply:
Still not sure I trust it :P

![image](https://github.com/user-attachments/assets/2ccf9713-c984-40c1-b360-44de97b94b0b)